### PR TITLE
Story/io 78/user roles 149

### PR DESCRIPTION
**Pull request by saijaseppa**
### Description 

Backend roles are implemented by ad_groups users have. 

Every user group has it's own permissions to read, modify and delete. 

Function user_in_viewer_group checks if user has correct group given. 
Function has_permission defines reading permissions to user. 
Function has_object_permission defines all permissions to create, update or delete anything. 

(Ticket related)[https://helsinkisolutionoffice.atlassian.net/jira/software/c/projects/IO/boards/302?selectedIssue=IO-57&assignee=63d13901ed33fd0707b8ee20]

### Known issues
When developing this the group "katselijat" ("viewer") was not yet given to my ext-account so testing that was only partial. This should be tested once the viewer group will be given.

### CODE REVIEW

File: infraohjelmointi_api/services/__init__.py [hunk 1]
```diff
@@ -16,3 +16,4 @@
 from .LocationFinancialService import LocationFinancialService
 from .SapApiService import SapApiService
 from .SapCostService import SapCostService
+from .ADGroupService import ADGroupService
```
Comment (millaeerola) [inline infraohjelmointi_api/services/__init__.py:19]: also unused here

File: infraohjelmointi_api/permissions.py [hunk 2]
```diff
@@ -1,6 +1,432 @@
+import logging
+from infraohjelmointi_api.models.Project import Project
+from infraohjelmointi_api.models.ProjectClass import ProjectClass
+from infraohjelmointi_api.models.ProjectGroup import ProjectGroup
 from rest_framework import permissions
+from infraohjelmointi_api.services.ADGroupService import ADGroupService
```
Comment (millaeerola) [inline infraohjelmointi_api/permissions.py:6]: This is an unused import

**Review by millaeerola — CHANGES_REQUESTED** [review]

File: infraohjelmointi_api/permissions.py [hunk 3]
```diff
@@ -1,6 +1,432 @@
...
 
+logger = logging.getLogger("infraohjelmointi_api")
```
[the remark this answers is not in the document]
Comment (saijaseppa) [reply infraohjelmointi_api/permissions.py:8]: Good point, deleted this. 👍

File: infraohjelmointi_api/serializers/WhoAmISerializer.py [hunk 4]
```diff
@@ -1,9 +1,11 @@
 from infraohjelmointi_api.models import User
 from rest_framework import serializers
+from infraohjelmointi_api.services.ADGroupService import ADGroupService
```
[the remark this answers is not in the document]
Comment (saijaseppa) [reply infraohjelmointi_api/serializers/WhoAmISerializer.py:3]: Deleted this.

**Pull request merged**